### PR TITLE
Parse SubjectAlternativeNames from CSR into Lemur Certificate

### PR DESCRIPTION
**Pull request by jramosf**
@castrapel I have adapted it for working with any.

Since cryptography.x509 classes do correspond to Lemur's API, the generalization is quite straightforward

### CODE REVIEW

**Comment by hosseinsh** [comment]
Javier, it would be really great, if you could also add automated testing for parsing SANs. This would make the integration easier.
Thanks

**Comment by jramosf** [comment]
> Javier, it would be really great, if you could also add automated testing for parsing SANs. This would make the integration easier.
> Thanks

Hi @hosseinsh, I've added `test_certificate_input_schema_parse_csr` to cover the new functionality.

Since we don't supply complete dicts from the tests to API and Python dicts don't have autovivification built-in, I've changed `schemas.py` accordingly. I've detected this edge case from running the new test.

Regards,

File: lemur/certificates/schemas.py [hunk 1]
```diff
@@ -112,10 +112,20 @@ def load_data(self, data):
...
+                data['extensions'] = {
+                    'subAltNames': {
+                        'names': []
+                    }
+                }
+            elif not data['extensions'].get('subAltNames'):
+                data['extensions']['subAltNames'] = {
+                    'names': []
+                }
+            elif not data['extensions']['subAltNames']['names']:
```
Comment (castrapel) [inline lemur/certificates/schemas.py]: `elif not data['extensions']['subAltNames'].get('names'):`

Comment (jramosf) [reply lemur/certificates/schemas.py]: I've added 58dd424de8781e852fbb4fac102273746120e5e2 to prevent this case, thanks!

**Comment by castrapel** [comment]
This looks great! Thanks for the PR, I'll merge it. Tested by going through certificate creation flow, including CSRs, and setting breakpoints at modified code.
